app/demo-loop/demo-loop.cc

```diff
@@ -62,6 +62,84 @@ using namespace celeritas;
 
 namespace
 {
+//---------------------------------------------------------------------------//
+//! `RootStepWriterFitler` helper function.
+bool rsw_filter_match(size_type step_attr_id, size_type filter_id)
+{
+    return filter_id == MCTruthFilter::unspecified()
+           || step_attr_id == filter_id;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * `RootStepWriter` filter.
+ *
+ * Write if event ID matches (or is undefined) *and* either the track ID or
+ * parent ID matches. If no fields are specified or are set to -1, all steps
+ * are stored.
+ */
+std::function<bool(RootStepWriter::TStepData const&)>
+make_root_step_writer_filter(LDemoArgs const& args)
+{
+    std::function<bool(RootStepWriter::TStepData const&)> rsw_filter;
+
+    if (args.mctruth_filter)
+    {
+        rsw_filter = [opts = args.mctruth_filter](
+                         RootStepWriter::TStepData const& step) {
+            return (rsw_filter_match(step.event_id, opts.event_id)
+                    && (rsw_filter_match(step.track_id, opts.track_id)
+                        || rsw_filter_match(step.parent_id, opts.parent_id)));
```

Comment:
If either `track_id` or `parent_id` is unspecified this will write the steps for _all_ tracks for the given event, correct? I think if only one of `track_id` or `parent_id` is given we still want to write _only_ if that ID is a match.

Reply:
Done. Now you can select any individual track, say `"mctruth_filter": { "event_id": 1, "track_id": 275 }`, leaving `parent_id` off and it will filter only steps for that specific track.
Of course, if a different filter logic is needed we can either update `demo-loop.cc::make_root_step_writer_filter()` or just create a new one to be passed to `RootStepWriter`.

Reply:
I think the logic error here was mine... I wanted to be able to specify both a parent and a track ID so that if you have a track that started with a weird state you could find out where it came from.

Reply:
@sethrj that would be useful too.. but then wouldn't we want to specify multiple track IDs rather than track ID + parent ID?

Reply:
Exactly 😂 I realized my stupidity a few minutes ago